cincinnati-client/cincinnati_client.go

```diff
@@ -46,6 +47,28 @@ type Release struct {
 	AvailableUpgrades []string
 }
 
+// SortAvailableUpgrades orders AvailableUpgrades in ascending semantic-version order.
+// It returns an error if any entry is not a valid semantic version.
+//
+// NOTE: Every version string must already be a valid semver, since they
+//
+//	are converted to semver objects when reading the cincinnati graph.
+func (r Release) SortAvailableUpgrades() error {
+	for i, upgrade := range r.AvailableUpgrades {
+		_, err := semver.NewVersion(upgrade)
+		if err != nil {
+			return fmt.Errorf("invalid semantic version in AvailableUpgrades[%d]=%q: %w", i, upgrade, err)
```

Comment:
super nitpick: not sure this error is helpful to debug anything with, maybe add the Version/channel it originates from, along the error message?

Reply:
added `r.Version` to the err msg. ptal